virttest/libvirt_xml/devices/disk.py

```diff
@@ -636,77 +628,27 @@ def __init__(self, virsh_instance=base.base.virsh):
                 super(self.__class__, self).__init__(virsh_instance=virsh_instance)
                 self.xml = '<source/>'
 
-    class DriverMetadata(base.base.LibvirtXMLBase):
-
+    class MetadataCache(base.base.LibvirtXMLBase):
         """
-        DriverMetaData XML class
-
-        <driver name='qemu' type='qcow2'>
-         <metadata_cache>
-           <max_size unit='bytes'>10</max_size>
-         </metadata_cache>
-        </driver>
-
-        Properties:
-
-        attrs: Dictionary of attributes, qualifying the driver name and type,etc.
-        metadata_cache: embedded class to describe metadata attributes
-        usages:
-            custom_disk = Disk(type_name='disk')
-            driver_dict = {"name": "qemu","type": "qcow2"}
-            new_one = custom_disk.new_drivermetadata(**{"attrs": driver_dict})
+        Source of MetadataCache xml class
 
-            metadata_cache_dict = {"max_size": "10", "max_size_unit": "bytes"}
-
-            new_one.metadata_cache = custom_disk.DriverMetadata().new_metadatacache(**metadata_cache_dict)
-
-            custom_disk.drivermetadata = new_one
+        max_size:
+            Int, attribute of MetadataCache max size
+        max_size_unit:
+            string, attribute of MetadataCache max size unit
         """
 
-        __slots__ = ('attrs', 'metadata_cache')
+        __slots__ = ('max_size', 'max_size_unit')
```

Comment:
<driver name='qemu' type='qcow2'>
driver itself can have attributes like above, how can we set this in MetadataCache class

Reply:
Moreover, from xml point view, metadata_cache is better to be defined as separate nested element in  driver tag

Reply:
If IIUC, the driver attributes will be set by using accessors.XMLElementDict('driver', self, parent_xpath='/',                                  tag_name='driver'),  the dict method, so we do not need to set them in  MetadataCache class. MetadataCache class is only for setting elements/attributes for  like below part:  
```
  <metadata_cache>
      <max_size unit='bytes'>1234</max_size>
    </metadata_cache>  
```

Reply:
@dzhengfy 
driver_metadatacache represents complete entity with driver and metadatache, why not open API to set driver attribute?

Reply:
> Moreover, from xml point view, metadata_cache is better to be defined as separate nested element in driver tag

@chunfuwen It's better only if 'driver' tag hasn't been defined for almost a decade and hasn't been referenced hundreds of times. 
We cannot afford to make such a tremendous change of code which could bring in unpredictable risks just for a slight advantage that could be ignored for most cases. It's just uneconomic.